project_task_code/models/project_task.py

```diff
@@ -41,3 +42,10 @@ def name_get(self):
             name = "[{}] {}".format(rec.code, task[1])
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        new_args = expression.OR([args or [], [("code", operator, name)]])
+        return super()._name_search(name, new_args, operator, limit, name_get_uid)
```

Comment:
```suggestion
    _rec_names_search = ["name", "code"]
```
You don't need to override `_name_search` anymore in v16.0.

cf https://github.com/odoo/odoo/pull/86588/commits/04c6859864c5ba229c1f443d7269588e348b2808